runtime/Go/antlr/atn_simulator.go

```diff
@@ -25,3 +32,15 @@ func (b *BaseATNSimulator) getCachedContext(context PredictionContext) Predictio
 
 	return getCachedBasePredictionContext(context, b.sharedContextCache, visited)
 }
+
+func (b *BaseATNSimulator) SharedContextCache() *PredictionContextCache {
+	return b.sharedContextCache
+}
+
+func (b *BaseATNSimulator) ATN() *ATN {
+	return b.atn
+}
+
+func (b *BaseATNSimulator) DecisionToDFA() []*DFA {
+	return b.decisionToDFA
+}
```

Comment:
Do the implementations need to be exported? Can we hide them behind New* funcs? This was something I was going to look into as well.